src/map.cpp

```diff
@@ -5936,10 +5953,13 @@ void map::vertical_shift( const int newz )
         return;
     }
 
-    clear_vehicle_cache();
-    vehicle_list.clear();
-    set_transparency_cache_dirty();
-    set_outside_cache_dirty();
+    // TODO: Remove the function when it's safe
+    return;
```

Comment:
My character does not move vertically anymore, probably because this prevents `m.abs_sub.z` being updated. The character is teleported (but only along the X,Y axis) to where the stairs on the other z-level are, but the actual z-level stays the same.

Not updating `m.abs_sub.z` means `game::get_levz()` will still return the old z-level, so the statement `u.setz( get_levz() );` (in `game::vertical_move`) does not do anything at all.